test/widgets/compose_box_test.dart

```diff
@@ -105,4 +118,70 @@ void main() {
       });
     });
   });
+
+  group('ComposeBox textCapitalization', () {
+    final message = eg.streamMessage();
+
+    Future<void> setupToComposeBox(WidgetTester tester, Narrow narrow) async {
+      addTearDown(testBinding.reset);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+        streams: [eg.stream(streamId: message.streamId)],
+      ));
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      final connection = store.connection as FakeApiConnection;
+
+      // prepare message list data
+      connection.prepare(json: GetMessagesResult(
+        anchor: message.id,
+        foundNewest: true,
+        foundOldest: true,
+        foundAnchor: true,
+        historyLimited: false,
+        messages: [message],
+      ).toJson());
+
+      await tester.pumpWidget(
+        MaterialApp(
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          home: GlobalStoreWidget(
+            child: PerAccountStoreWidget(
+              accountId: eg.selfAccount.id,
+              child: MessageListPage(narrow: narrow)))));
```

Comment:
Instead of `MessageListPage`, this can use just `ComposeBox` directly.

That also simplifies the setup — e.g. it means we no longer need to prepare a `GetMessagesResult`.)